### PR TITLE
Etcd

### PR DESCRIPTION
**Pull request by toolslive**
suite "kwik" is ok. still have to validate the others, but I don't expect any problems.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -64,6 +66,8 @@ install_client:
           _build/src/tools/typed_ssl.mli \
           _build/src/tools/typed_ssl.cmi \
           _build/src/tools/unix_fd.cmi \
+	  _build/src/tools/arakoon_etcd.cmi \
+	  _build/src/tools/arakoon_url.cmi \
```
Comment (domsj) [inline Makefile:70]: indented using a tab iso spaces ... I though Makefiles didn't like that?

Comment (toolslive) [inline Makefile:70]: `Each line in the recipe must start with a tab` from
https://www.gnu.org/software/make/manual/html_node/Recipe-Syntax.html

File: src/tools/arakoon_url.ml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+limitations under the License.
+*)
+
+type url =
+  File of string
+  | Etcd of (((string * int) list) * string)
+
+let to_string = function
+  | File f -> f
+  | _ -> "??"
```
Comment (domsj) [inline src/tools/arakoon_url.ml]: Shouldn't be too hard to give a good string representation here?
sth a la `etcd://host1:port1,host2:port2/key`

File: src/node/node_cfg.ml [hunk 3]
```diff
@@ -612,127 +612,129 @@ module Node_cfg = struct
...
-
-  let read_client_config fn = to_client_cfg (read_config fn)
+                                       [] (inifile # sects) in
+      let () = if List.length remaining > 0 then
+                 failwith ("Can't find config section for: " ^ (String.concat "," remaining))
+      in
+      let quorum_function = _get_quorum_function inifile in
+      let lease_period = _get_lease_period inifile in
+      let cluster_id = _get_cluster_id inifile in
+      let m_n_cfg = None (* TODO: get_nursery_cfg inifile config_file *) in
```
Comment (domsj) [inline src/node/node_cfg.ml]: This breaks the nursery functionality?

File: src/nursery/client_cfg.ml [hunk 4]
```diff
@@ -84,4 +84,17 @@ module ClientCfg = struct
...
+  let _from_file section filename =
+    Lwt_extra.read_file filename >>= fun txt ->
+    _from_txt section txt |> Lwt.return
+
+  let _from_etcd section peers path =
+    Arakoon_etcd.retrieve_value peers path >>= fun txt ->
+    _from_txt section txt |> Lwt.return
+
+  let from_url section url =  (* This is the format as defined in the extension *)
+    match url with
```
Comment (domsj) [inline src/nursery/client_cfg.ml]: This pattern match should really be done only once. (Now I see it three times in the pull request)

**Comment by domsj** [comment]
Ok

**Pull request merged**